### PR TITLE
sip: add tests for sip over tcp

### PR DESCRIPTION
**Pull request by glongo**
This adds tests for sip over tcp.

Update:
- Added C code for generating a pcap sample.

Redmine ticket
https://redmine.openinfosecfoundation.org/issues/3351

### CODE REVIEW

**Comment by glongo** [comment]
Suricata PR: https://github.com/OISF/suricata/pull/9321

File: tests/sip-tcp-method/test.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+requires:
+  features:
+    - HAVE_LIBJANSSON
+  min-version: 5.0.0
```
Comment (catenacyber) [inline tests/sip-tcp-method/test.yaml:4]: Should be 7 (and for other tests)

File: tests/sip-tcp-protocol/test.rules [hunk 2]
```diff
@@ -0,0 +1,2 @@
+alert sip any any -> any any (flow:to_server; sip.protocol; content:"SIP/2.0"; sid:1;)
+alert sip any any -> any any (flow:to_client; sip.protocol; content:"SIP/2.0"; sid:2;)
```
Comment (catenacyber) [inline tests/sip-tcp-protocol/test.rules:2]: I wonder if it makes more sense to have only one test sip-tcp-keywords...

File: tests/sip-tcp-protocol/test.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+requires:
+  features:
+    - HAVE_LIBJANSSON
```
Comment (catenacyber) [inline tests/sip-tcp-protocol/test.yaml:3]: You do not need to put this with min-version 7 ;-)

**Pull request closed without merging**